cmd/kube-bench-summarizer/summarizer/summarizer.go

```diff
@@ -0,0 +1,200 @@
+	State       kb.State               `json:"status"`
+	ActualValue string                 `json:"-"`
+	Scored      bool                   `json:"-"`
+	Hosts       []string               `json:"hosts"`
+}
+
+type GroupWrapper struct {
+	ID     string          `yaml:"id" json:"section"`
+	Pass   int             `json:"pass"`
+	Fail   int             `json:"fail"`
+	Warn   int             `json:"warn"`
+	Info   int             `json:"info"`
+	Text   string          `json:"desc"`
+	Checks []*CheckWrapper `json:"results"`
+}
+
+type SummarizedReport struct {
+	Version string          `json:"-"`
+	Groups  []*GroupWrapper `json:"tests"`
+	kb.Summary
+}
+
+func NewSummarizer(inputDir, outputDir string) *Summarizer {
+	return &Summarizer{
+		InputDirectory:  inputDir,
+		OutputDirectory: outputDir,
+		summarizedReport: &SummarizedReport{
+			Groups: []*GroupWrapper{},
+		},
+		groupsMap:  map[string]*GroupWrapper{},
+		checksMaps: map[string]*CheckWrapper{},
+	}
+}
+
+func (s *Summarizer) summarizeForHost(hostname string) error {
+	logrus.Debugf("summarizeForHost: %v", hostname)
+
+	hostDir := fmt.Sprintf("%v/%v", s.InputDirectory, hostname)
+	resultFiles, err := filepath.Glob(fmt.Sprintf("%v/*.json", hostDir))
+	if err != nil {
+		return err
+	}
+
+	for _, fileName := range resultFiles {
+		// Load one result file
+		// Marshal it into the results
+		contents, err := ioutil.ReadFile(filepath.Clean(fileName))
+		if err != nil {
+			logrus.Errorf("error reading file %+v: %v", fileName, err)
+			return err
+		}
+
+		results := &kb.Controls{}
+		err = json.Unmarshal(contents, results)
+		if err != nil {
+			logrus.Errorf("error unmarshalling: %v", err)
+			return err
+		}
+		logrus.Debugf("results: %+v", results)
+
+		for _, group := range results.Groups {
+			for _, check := range group.Checks {
+				logrus.Infof("id: %v %v", check.ID, check.State)
+				if check.Scored && check.State != kb.PASS {
+					logrus.Infof("check: %+v", check)
+					g := s.groupsMap[group.ID]
+					if g == nil {
+						logrus.Infof("group not found in map: %v", group.ID)
+						g = &GroupWrapper{
+							ID:     group.ID,
+							Text:   group.Text,
+							Checks: []*CheckWrapper{},
+						}
+						s.groupsMap[group.ID] = g
+						s.summarizedReport.Groups = append(s.summarizedReport.Groups, g)
+					}
+
+					c := s.checksMaps[check.ID]
+					if c == nil {
+						logrus.Infof("check not found in map: %v", check.ID)
+						c = &CheckWrapper{
+							ID:          check.ID,
+							Text:        check.Text,
+							Remediation: check.Remediation,
+							TestInfo:    check.TestInfo,
+							State:       check.State,
+							Hosts:       []string{},
+						}
+						s.checksMaps[check.ID] = c
+						g.Checks = append(g.Checks, c)
+
+						switch check.State {
+						case kb.FAIL:
+							g.Fail++
+							s.summarizedReport.Fail++
+						case kb.WARN:
+							g.Warn++
+							s.summarizedReport.Warn++
+						case kb.INFO:
+							g.Info++
+							s.summarizedReport.Info++
```

Comment:
shall we fall back to some default?

Reply:
There is no other state, but adding a default doesn't hurt.